tracing/zipkin.go

```diff
@@ -34,3 +37,38 @@ func CreateZipkinReporter(cfg *config.Config) (zipkinreporter.Reporter, error) {
 	}
 	return httpreporter.NewReporter(cfg.ZipkinEndpoint), nil
 }
+
+func WithZipkinExporter(reporterFact ZipkinReporterFactory, endpoint *zipkinmodel.Endpoint) ConfigOption {
+	return func(cfg *config.Config) {
+		var (
+			reporter zipkinreporter.Reporter
+			exporter trace.Exporter
+		)
+
+		if cfg != nil && cfg.Enable {
+			// Initialize our reporter / exporter
+			// do this before cleanup to minimize time where we have duplicate exporters
+			reporter, err := reporterFact(cfg)
+			if err != nil {
+				// TODO(greghaynes) log this error
+				return
+			}
+			exporter := zipkin.NewExporter(reporter, endpoint)
+			trace.RegisterExporter(exporter)
+		}
+
+		// We know this is set because we are called with acquireGlobal lock held
+		oct := globalOct
+		if oct.exporter != nil {
+			trace.UnregisterExporter(oct.exporter)
+		}
+
+		if oct.closer != nil {
+			// TODO(greghaynes) log this error
```

Comment:
Not a an issue with this PR so you can leave it, but we should remove this comment